test/core/ext/filters/event_engine_client_channel_resolver/resolver_ops.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package event_engine_client_channel_resolver;
+
+message ResolverArgs {
+    bool request_service_config = 2;
+    bool enable_srv_queries = 3;
+}
+
+// Some error. The details are unimportant.
+message Error {}
+
+
+message ResolvedAddress {
+    uint32 port = 1;
+}
+message ResolvedAddresses {
+    repeated ResolvedAddress addresses = 1;
+}
+
+message SRVRecord {
+    string host = 1;
+    uint32 port = 2;
+    uint32 priority = 3;
+    uint32 weight = 4;
+}
+message SRVRecords {
+    repeated SRVRecord srv_records = 1;
+}
+
+enum TXTRecordType {
+    TXT_VALID = 0;
+    TXT_INVALID = 1;
+}
+message TXTRecords {
+    repeated TXTRecordType txt_records = 1;
+}
+
+message Msg {
```

Comment:
Might be useful to also support re-resolution, and to test the backoff in case of a resolver failure.

Reply:
I think that's doable. IIUC, that amounts to calling the result health callback with bad statuses some number of times (capped at maybe 3, since that's a ~7s test already) before succeeding. We'd see timeouts if re-resolution failed.